File: 872-leaf-similar-trees/872-leaf-similar-trees.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        ArrayList<Integer> list1 = new ArrayList<>();
        helper(root1, list1);
        ArrayList<Integer> list2 = new ArrayList<>();
        helper(root2, list2);
        return list1.equals(list2);
    }
    
    public static void helper(TreeNode root, ArrayList<Integer> list){
        if(root == null) return;
        if(root.left==null && root.right == null){
            list.add(root.val);
        }
        helper(root.left, list);
        helper(root.right, list);
    }
}

File: 872-leaf-similar-trees/872-leaf-similar-trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
var leafSimilar = function(root1, root2) {
    let l1 = []
    let l2 = []
    helper(root1, l1)
    helper(root2, l2)
    return JSON.stringify(l1)===JSON.stringify(l2);
};

var helper = function(root, list){
        if(root == null) return;
        if(root.left==null && root.right == null){
            list.push(root.val);
        }
        helper(root.left, list);
        helper(root.right, list);
    }















